### PR TITLE
Empty fields exclusion in safe attributes of Model

### PR DESCRIPTION
**Pull request by manchenkoff**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #15557 

This change will exclude empty fields like `['' => 'value']` from safe attributes.

### CODE REVIEW

File: framework/base/Model.php [hunk 1]
```diff
@@ -790,7 +790,7 @@ public function safeAttributes()
         }
         $attributes = [];
         foreach ($scenarios[$scenario] as $attribute) {
-            if (strncmp($attribute, '!', 1) !== 0 && !in_array('!' . $attribute, $scenarios[$scenario])) {
+            if (mb_strlen($attribute) !== 0 && strncmp($attribute, '!', 1) !== 0 && !in_array('!' . $attribute, $scenarios[$scenario])) {
```
Comment (bizley) [inline framework/base/Model.php:793]: strlen is sufficient here

**Review by bizley — APPROVED** [review]

File: framework/base/Model.php (hunk 1 above)
Comment (manchenkoff) [reply framework/base/Model.php]: changed to strlen()

**Comment by WinterSilence** [comment]
`$attribute !== ''` better, but I think is redundant check - errors like this must be detected at developing/testing.

**Review by Arhell — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**